tools/test/audio/comp_run.sh

```diff
@@ -152,17 +154,31 @@ else
     VALGRIND_CMD=
 fi
 
-HOST_EXE="$XTRUN_CMD $TESTBENCH"
+HOST_EXE="$XTRUN_CMD $PATH_TESTBENCH"
 
 # Use topology from component test topologies
 INFMT=s${BITS_IN}le
 OUTFMT=s${BITS_OUT}le
 TPLGFN=test-${DIRECTION}-ssp5-mclk-0-I2S-${COMP}-${INFMT}-${OUTFMT}-48k-24576k-codec.tplg
+TPLG_BUILD_TIP="Please run scripts/build-tools.sh -t"
+PIPELINES=
+
+[[ $TESTBENCH == "sof-testbench4" ]] && {
+    # With comp benchmark topologies for playback use pipelines 1-2, for capture 3-4
+    [[ $DIRECTION == "playback" ]] && PIPELINES="-p 1,2"
+    [[ $DIRECTION == "capture" ]] && PIPELINES="-p 3,4"
+    TPLGFN=sof-hda-benchmark-${COMP}${BITS_IN}.tplg
+    TPLG_DIR="../../build_tools/topology/topology2/development"
+    TPLG_BUILD_TIP="Please run scripts/build-tools.sh"
+    # Hack, remove this and enable valgrind
+    VALGRIND_CMD=
```

Comment:
Still a valgrind issue found with controls implementation that causes all tests to fail, need to fix.